backend/matching-service/app/src/utils/preferences.ts

```diff
@@ -0,0 +1,16 @@
+import _ from "lodash";
+import { Preferences } from "../matching/matching.interfaces";
+
+export const comparePreferences = (p1: Preferences, p2: Preferences): boolean => {
+  // Check if difficulty matches
+  if (_.isEqual(p1.difficulty, p2.difficulty)) {
+    // Check if object1's topics are a subset of or equal to object2's topics and vice versa
+    if (
+      _.intersection(p1.category, p2.category).length === p1.category.length ||
+      _.intersection(p1.category, p2.category).length === p2.category.length
+    ) {
```

Comment:
```suggestion
    if (
      (_.intersection(p1.category, p2.category).length === p1.category.length) ||
     ( _.intersection(p1.category, p2.category).length === p2.category.length)
    ) {
```

could you help to add extra brackets JUST in case the order of operations is wrong

e.g. if the order of operations is interpreted in this way, this would cause bugs
```
if (a === b || c === d)
```
becomes interpreted as
```
if ((a === b || c) === d)
```

Reply:
It won't be wrong haha equal is always higher precedence than or according to this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_precedence